### PR TITLE
`build/pkgs/flint/spkg-configure.m4`: Reject broken FLINT package on homebrew

### PR DESCRIPTION
**Pull request by mkoeppe**
Fixes https://github.com/sagemath/sage/pull/38020#issuecomment-2118840044

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: build/pkgs/flint/spkg-configure.m4 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 SAGE_SPKG_CONFIGURE([flint], [
     SAGE_SPKG_DEPCHECK([mpfr], [
-        AC_CHECK_HEADER(flint/flint.h, [dnl
+        AC_CHECK_HEADERS([flint/flint.h flint/padic.h], [dnl
           dnl gr_get_fexpr appears in Flint 3.0
```
Comment (kwankyu) [inline build/pkgs/flint/spkg-configure.m4:4]: Do you want to check `padic_types.h` or `padic.h`?

Comment (mkoeppe) [reply build/pkgs/flint/spkg-configure.m4:4]: intended as done here

Comment (kwankyu) [reply build/pkgs/flint/spkg-configure.m4:4]: On my system:
```
flint$ ls -l padic*
-rw-r--r--  1 kwankyu  admin   8583 Feb 26 01:47 padic.h
-rw-r--r--  1 kwankyu  admin   6365 Feb 26 01:47 padic_mat.h
-rw-r--r--  1 kwankyu  admin  11871 Feb 26 01:47 padic_poly.h
flint$ pwd
/usr/local/Cellar/flint/3.1.0/include/flint
```
So `padic_types.h` is the missing one. No?

Comment (mkoeppe) [reply build/pkgs/flint/spkg-configure.m4:4]: checking for header files that the Sage library uses directly.

Comment (kwankyu) [reply build/pkgs/flint/spkg-configure.m4:4]: I see. `AC_CHECK_HEADERS` checks if `flint/padic.h` works correctly. OK.

**Review by kwankyu — APPROVED** [review]
Works well.

File: build/pkgs/flint/spkg-configure.m4 (hunk 1 above)
Comment (mkoeppe) [reply build/pkgs/flint/spkg-configure.m4:4]: That's right, it does so by compiling a test program.

**Comment by mkoeppe** [comment]
Thank you!

**Comment by edgarcosta** [comment]
Did I do something wrong in [flint 3.1.3](https://github.com/Homebrew/homebrew-core/pull/166490) ?

**Comment by mkoeppe** [comment]
```
brew reinstall flint                                    git:t/32432/modularization_of_sagelib__break_out_a_separate_package_sagemath_polyhedra
==> Downloading https://ghcr.io/v2/homebrew/core/flint/manifests/3.1.0
Already downloaded: /Users/mkoeppe/Library/Caches/Homebrew/downloads/deb933acbf833778d6ddfabb9aa1b10c2f9e8f6bc74e477e21bd8094fa9a3005--flint-3.1.0.bottle_manifest.json
==> Fetching flint
==> Downloading https://ghcr.io/v2/homebrew/core/flint/blobs/sha256:38b55243359af6245acad6bb3135eb23d3502ac5d4bf53249a0f383f2f1b9911
Already downloaded: /Users/mkoeppe/Library/Caches/Homebrew/downloads/efa54108f40bef34318257abc445b586447e189c1594871f6cd4805f9189ff19--flint--3.1.0.sonoma.bottle.tar.gz
==> Reinstalling flint 
==> Pouring flint--3.1.0.sonoma.bottle.tar.gz
🍺  /usr/local/Cellar/flint/3.1.0: 178 files, 22.4MB
==> Running `brew cleanup flint`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> No outdated dependents to upgrade!
==> Checking for dependents of upgraded formulae...
Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Reinstalling 1 dependent with broken linkage from source:
arb
Error: arb has been disabled because it arb has been merged into flint 3.0.0!
worktree-clean ➤ ls /usr/local/opt/flint/include/flint/padic*            git:t/32432/modularization_of_sagelib__break_out_a_separate_package_sagemath_polyhedra
/usr/local/opt/flint/include/flint/padic.h      /usr/local/opt/flint/include/flint/padic_mat.h  /usr/local/opt/flint/include/flint/padic_poly.h
worktree-clean ➤ tar tfz /Users/mkoeppe/Library/Caches/Homebrew/downloads/efa54108f40bef34318257abc445b586447e189c1594871f6cd4805f9189ff19--flint--3.1.0.sonoma.bottle.tar.gz |grep padic
flint/3.1.0/include/flint/padic.h
flint/3.1.0/include/flint/padic_mat.h
flint/3.1.0/include/flint/padic_poly.h
```

**Comment by mkoeppe** [comment]
@edgarcosta 3.1.3 looks fine.
```
$ ls /usr/local/opt/flint/include/flint/padic* 
/usr/local/opt/flint/include/flint/padic.h       /usr/local/opt/flint/include/flint/padic_poly.h
/usr/local/opt/flint/include/flint/padic_mat.h   /usr/local/opt/flint/include/flint/padic_types.h
```

**Pull request merged**